### ISSUE TITLE
Moving Head Repositioning Widget

### ISSUE THREAD

**Issue by MCStoff** [xLightsSequencer/xLights#4784]
**Is your feature request related to a problem? Please describe.**
A significant portion of moving head sequencing could be automated with a widget in the moving head effect.  At present, the moving heads must be repositioned between effects by adding a command with the moving head shut off.  Pan and tilt values from the existing position are changed to the next position of the head.

**Describe the solution you'd like**
A drag and drop repositioning tool that would take the pan and tilt from position A found in the first effect and reposition the heads to position B found in the second effect.  Using the slew limits and the time between the two effects, an alert could be created notifying the sequencer when there is insufficient time to do the repositioning they sequenced given the limitations of their hardware.  The widget would place an effect between the first and second moving head effect that has a value curve starting with the pan position of the first effect and ending with the pan position of the second effect.  It would also include in that effect a value curve starting with the tilt position of the first effect and ending with the tilt position of the second effect.  The time available to execute the position change could be compared to the time necessary to execute the position change given the slew limits to generate the alerts.

**Describe alternatives you've considered**
At present, there are no alternatives other than the manual repositioning described above.  While in itself, the repositioning of the heads is not difficult, because the data has already been entered into the effects, it could be automated, thereby saving the end user significant time.

**Additional context**
The addition of the widget would allow the end user to focus on the visual portion of the moving heads without regard for the repositioning task.  Making it drag and drop would allow the end user to continue to manually enter in the repositioning should they desire to do so.  It also would not affect existing moving head sequences.  Including the alert would help sequencers avoid unintended results when there is insufficient time to reposition the moving head.

**Comment by cybercop23** [comment]
Yeah.. this is on "our" list to _fix_. The idea is mybe make it work like the new servo effect where you can link the end of the previous effect to the start of the new one and up to you to have enough time between for the heard to move. The original thought was a right click option to Insert MH Position to do just this.
For now, the easiest way is to split the effect that you have and drag the split (middle) to the right just passed the dimmer start. This was demoed by MarkB in last week's xLights Gems session.

**Comment by MCStoff** [comment]
Glad to hear that it is in the pipeline.  It seems like an obvious and doable enhancement.  Thank you for the information.

**Comment by derwin12** [comment]
Link feature in the next update.
